### DISCUSSION TITLE
Very very much of need help !Cannot read properties of null (reading 'useEffect')

### DISCUSSION THREAD

**Discussion by Athrun-Judah** [vercel/next.js#58571] [App Router]
![image](https://github.com/vercel/next.js/assets/66123941/6b6c46a9-9301-4e91-9f1f-ff159b34334e)
The version of next is 14.0.2.
please tell me how to solve this problem, now I can't build the project, very urgent!

**Comment by icyJoseph** [comment]
Got a repository to take a look at? Looks like you might have forgotten to add `use client` to a component that is using useEffect.

**Reply by Athrun-Judah** [reply]
This problem occurs when I use tanstack/query, not without adding "use client". But the author of tanstack/query said it is the problem of nextjs.

**Reply by icyJoseph** [reply]
> This problem occurs when I use tankstack/query, not without adding "use client"

What do you mean? Are you using `use client` or not... to use something like tanstack/query, you have to opt each module that uses an API from their library, with a `use client` directive.

**Reply by Athrun-Judah** [reply]
Yeah, this is what I did. Please look this: https://github.com/TanStack/query/issues/5419#issuecomment-1549891511

**Reply by Athrun-Judah** [reply]
There was no error when I used nextjs@13.4.9
